rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,62 @@
+---
+feature: use-matrix-for-official-chat
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from IRC ([Freenode](https://freenode.net/)) to
+[Matrix](https://matrix.org/)
+([#community:nixos.org][1]) as
+our official community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network effects may be negative instead of
+positive.
+
+There are currently 1796 members of the Unofficial NixOS Discord
+channel (244 weekly visitors, 95 weekly communicators). Some of them
+might be convinced by the more modern features available on Matrix to
+move their conversations there.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to our Matrix
+Space ([#community:nixos.org][1]) that has already been created which you
+can join by visting the [invite
+link][1].
+
+Update any other places that point to Freenode to point to Matrix
+instead.
+
+Port any bots that might be necessary such as:
+
+* nix-channel-monitor
+* github PR events
+* karma bot
+* eval bot
+* factoids bot
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+People are used to Freenode and it might be harder for people to use
+Matrix, which requires more complex clients.
+
```

Comment:
Presence data is much more chatty in Matrix and can be turned off on the server level. For large rooms, it might be a good idea to do that. Especially since it leaks some level of personal information. But it's tunable only on the server level the last time I looked into it.

Reply:
Presence is enabled on a per-server basis. If you do not want presence data, then you can use a server that doesn't support it. Matrix.org, for example, currently doesn't support presence notifications because they are so expensive computationally.

Reply:
@sumnerevans Both of you are correct:
You're talking about the explicit sending of presence as a *being available or not* flag being broadcasted. That can indeed be disabled.
@zimbatm is likely talking about the read state marker being moved forward, which can be used to infer user presence.

Reply:
Can presence not be turned off on a room level? Even if I turn it off on my server, the sync will still broadcast other people's presence and take up a lot of space.

Reply:
@zimbatm it depends on what kind of "presence" you are talking about. If you are talking about read state, then that's not possible to disable at all.

If you are talking about the "presence" feature (which shows users as available or unavailable in the UI), then that can be disabled on a per-server basis, and it will ignore all presence notifications coming over federation, and will also not send out any presence notifications.

Reply:
There is an open issue for disabling read reciepts: https://github.com/vector-im/element-web/issues/2527

Reply:
Support for MSC2285 (hidden read receipts) just landed in Synapse v1.40.0rc1. Here is the PR: https://github.com/matrix-org/synapse/pull/10413. And here are the release notes: https://github.com/matrix-org/synapse/releases/tag/v1.40.0rc1